sql-plugin/src/main/scala/com/nvidia/spark/rapids/RegexParser.scala

```diff
@@ -425,6 +433,11 @@ class CudfRegexTranspiler {
         case '.' =>
           // workaround for https://github.com/rapidsai/cudf/issues/9619
           RegexCharacterClass(negated = true, ListBuffer(RegexChar('\r'), RegexChar('\n')))
+        case '^' | '$' if replace =>
+          // this is a bit extreme and it would be good to replace with finer-grained
+          // rules
+          throw new RegexUnsupportedException("regexp_replace on GPU does not support ^ or $")
```

Comment:
I filed https://github.com/NVIDIA/spark-rapids/issues/4091 as a follow-on issue to improve support here